sonar-php-plugin/src/test/java/org/sonar/plugins/php/reports/phpunit/PhpUnitSensorTest.java

```diff
@@ -113,6 +118,56 @@ public void shouldUseMultiPathCoverage() throws IOException {
       .contains(resourcesFolder + "', nothing will be imported from this report.");
   }
 
+  @Test
+  public void shouldLogWarningWhenTestCaseIsDetectedWithoutDeclaration() {
+    Set.of("src/App.php", "src/EmailTest.php").forEach(
+      file -> context.fileSystem().add(inputFile(file))
+    );
+
+    Sensor sensor = createSensor();
+    sensor.execute(context);
+
+    assertThat(logTester.logs(LoggerLevel.WARN)).contains(EXPECTED_MESSAGE);
+    assertThat(logTester.logs(LoggerLevel.DEBUG)).hasSize(1);
+    assertThat(logTester.logs(LoggerLevel.DEBUG).get(0)).startsWith("Detected and undeclared test case in").endsWith("src/EmailTest.php");
+  }
+
+  @Test
+  public void shouldNotLogWarningWhenTestCaseIsDetectedWithDeclaration() {
+    Set.of("src/App.php", "src/EmailTest.php").forEach(
+      file -> context.fileSystem().add(inputFile(file))
+    );
+
+    context.settings().setProperty("sonar.tests", "tests");
```

Comment:
The directory set here and path `src/EmailTest.php` should be consistent - to be a more real case scenario.